feat(docs): serve for Firebase Cloud Functions (#841)
inngest/website: pages/docs/sdk/serve.mdx

You can run this locally with `npx @google-cloud/functions-framework --target=inngest` which will serve your Inngest functions on port `8080`.

## Framework: Firebase Cloud Functions

Based on the Google Cloud Function architecture, the Firebase Cloud Functions provide a different API to serve functions using `onRequest`:

```typescript
import { onRequest } from "firebase-functions/v2/https";

import { serve } from "inngest/express";
import { inngest as inngestClient } from "./inngest/client";

export const inngest = onRequest(
  serve({
    client: inngestClient,
    functions: [/* ...functions... */],
  })
);
```

<Callout>

> Firebase Cloud Function requires to configure the `INNGEST_SERVE_PATH` with the custom function path.

Reviewer: ```suggestion
Firebase Cloud Functions require configuring `INNGEST_SERVE_PATH` with the custom function path.
```

"Require" is a transitive verb that needs a direct object to complete its meaning when followed by an infinitive phrase. By changing the phrase to gerund, we are providing a direct object.

For example, for a project named `inngest-firebase-functions` deployed on the `us-central1` region, the `INNGEST_SERVE_PATH` value will be as follows:

```
/inngest-firebase-functions/us-central1/inngest/
```

</Callout>

> You can now server your Firebase Cloud Function locally using the following command:

Reviewer: ```suggestion
To serve your Firebase Cloud Function locally, use the following command:
```

"You can now" sounds conversational and given that this is not a tutorial but a reference page, we provide directions.

```bash
firebase emulators:start
```

> Please note that you'll need to start your Inngest Local Dev Server with the `-u` flag as follows:

Reviewer: Could you add a reason for that?

Author: Updated!

> Please note that you'll need to start your Inngest Local Dev Server with the `-u` flag as follows:

Author: ```suggestion
Please note that you'll need to start your Inngest Local Dev Server with the `-u` flag as follows to match our Firebase Cloud Function's custom path:
```

```bash
npx inngest-cli@latest dev -u http://127.0.0.1:5001/inngest-firebase-functions/us-central1/inngest
```

> _The above command example features a project named `inngest-firebase-functions` deployed on the `us-central1` region_

Reviewer: ```suggestion
_The above command example features a project named `inngest-firebase-functions` deployed on the `us-central1` region_.
```
